sockeye/model.py

```diff
@@ -317,6 +317,17 @@ def load_parameters(self,
                                 cast_dtype=cast_dtype, dtype_source=dtype_source)
         logger.info('Loaded params from "%s" to "%s"', filename, mx.cpu() if ctx is None else ctx)
 
+    def set_parameters(self, new_params: dict):
```

Comment:
Can you change the type to `Dict[str, mx.gluon.parameter.Parameter]` to make clear that this expects a mapping of Gluon parameters?